smac_planner/src/a_star.cpp

```diff
@@ -480,6 +638,36 @@ unsigned int & AStarAlgorithm<NodeT>::getSizeDim3()
   return _dim3_size;
 }
 
+template<typename NodeT>
+typename AStarAlgorithm<NodeT>::NodePtr AStarAlgorithm<NodeT>::tryAnalyticExpansion(
+  const NodePtr & current_node, const NodeGetter & getter, int & analytic_iterations,
+  int & closest_distance)
+{
+  if (_motion_model == MotionModel::DUBIN || _motion_model == MotionModel::REEDS_SHEPP) {
+    // This must be a NodeSE2 node if we are using these motion models
+
+    // Then check if it is collision free.
+    // Always run the expansion on the first run in case there is a
+    // trivial path to be found
+    if (analytic_iterations <= 0) {
+      // Reset the counter, and try the analytic path expansion
+      analytic_iterations = closest_distance;
+      return getAnalyticPath(current_node, getter);
+    }
+    analytic_iterations--;
+    // See if we are closer and should be expanding more often
+    const Coordinates node_coords =
+      NodeT::getCoords(current_node->getIndex(), getSizeX(), getSizeDim3());
+    closest_distance =
+      std::min(
+      closest_distance,
+      static_cast<int>(NodeT::getHeuristicCost(node_coords, _goal_coordinates)/NodeSE2::neutral_cost));
```

Comment:
`NodeT::neutral_cost`, since this is a templated function

Reply:
Fixed